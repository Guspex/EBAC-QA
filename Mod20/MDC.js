export function mdc(num1, num2) {

    var resto;

    do {
        resto = num1 % num2;

        num1 = num2;
        num2 = resto;

    } while (resto != 0);
    console.log(num1);
    return num1;
}