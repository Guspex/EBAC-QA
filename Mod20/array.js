let array = [18,5,6,8,4,32,48,9,19,46,75,15,20,35]
let indiceMaior = 0
let indiceMenor = 0
for (var i = 0; i < array.length; i++) {
    for (var j = 0; j < array.length; j++) {
        let temp
        if (array[i] > array[j]) {
            temp = i
            if (array[indiceMaior] < array[temp]){
                indiceMaior = temp
            }
        }
        if (array[i] < array[j]) {
            temp = i
            if (array[indiceMenor] > array[temp]){
                indiceMenor = temp
            }
        }
    }
}

console.log(indiceMaior)
console.log(indiceMenor)
let indices = [indiceMaior, indiceMenor]
export default indices