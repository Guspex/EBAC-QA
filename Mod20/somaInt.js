let totalDaSoma = 0
function getTotalDaSoma() {
for (var i = 0; i < 1000; i++) {
    if (i % 5 == 0 || i % 7 == 0) {
        totalDaSoma += i;
    }
}

return totalDaSoma
}
console.log(getTotalDaSoma())
export default totalDaSoma