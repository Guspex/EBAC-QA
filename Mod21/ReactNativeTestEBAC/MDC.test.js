import {expect} from '@jest/globals';
import React from "react";
import renderer from "react-test-renderer";
import {mdc} from "../../Mod20/MDC"


test('This is validated', () => {
    const tree = renderer.create(<mdc />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('That result is equal', () => {
    expect(mdc(10,20)).toEqual(10);
})
