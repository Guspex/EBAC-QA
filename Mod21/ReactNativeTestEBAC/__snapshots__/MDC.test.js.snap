// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This is validated 1`] = `<mdc />`;
